File: Recursion/Factorial/Factorial.py
number = int(input("Enter the number: "))

def factorial(num):
    if num == 0:
        return 1
    return num * factorial(num-1)

if number < 0:
    print("Error")
else:
    print("Factorial = ", factorial(number))

    # time complexity: O(n)
    # space complexity: O(n)

File: Recursion/Factorial/factorial.java
import java.util.Scanner;

class Factorial {

    static int factorial(int n) {
        if (n == 0) {
            return 1;
        }
        return n * factorial(n-1);
    }
    public static void main(String args[]) {
        System.out.print("Enter a number: ");
        Scanner input = new Scanner(System.in);
        int number = input.nextInt();
        System.out.print("Factorial of "+ number + " is " + factorial(number));
        input.close();
    }
}